dace/transformation/transformation.py

```diff
@@ -107,29 +152,49 @@ def __init__(self, sdfg_id, state_id, subgraph, expr_index):
         # Serializable subgraph with node IDs as keys
         expr = self.expressions()[expr_index]
         self._subgraph = {expr.node_id(k): v for k, v in subgraph.items()}
-        self._subgraph_user = subgraph
+        self._subgraph_user = copy.copy(subgraph)
         self.expr_index = expr_index
 
+        # Ease-of-use API: Set new pattern-nodes with information about this
+        # instance.
+        for pname, pval in self._get_pattern_nodes().items():
+            # Create new pattern node from existing field
+            new_pnode = PatternNode(
```

Comment:
why PatternNode is not inside the module name?

Reply:
Because it is defined in the same file.